### ISSUE TITLE
Pyright complaint about type issue when passing to *args

### ISSUE THREAD

**Issue by T0maas** [microsoft/pyright#3842]
**Describe the bug**
This message appears when I try to assign some type to *args function parameter.

**To Reproduce**
This is part of my code:
```python                                                                                                                                                                       
  import gi                                                                                                                                                                                                                                                                                                                     
  gi.require_version("Gtk", "3.0")                                                                                                                                     
  gi.require_version("Gdk", "3.0")                                                                                                                                     
  from gi.repository import GObject, Gdk, Gtk                                                                                                                          
      
  builder = Gtk.Builder()                                                                                                                                              
  builder.add_from_file("ui.glade")  # Here it complains
```
The message is: `[Pyright reportGeneralTypeIssues] [E] Argument of type "Literal['ui.glade']" cannot be assigned to parameter "args" of type "Builder" in function "add_from_file"   "Literal['ui.glade']" is incompatible with "Builder" ...`

Code runs correctly, only pyright complains.

**Comment by erictraut** [comment]
What package do I need to install for the import module `gi`? Which version do you have installed? Is it a typed library? If not, do you have type stubs for it?

If you right click on `add_from_file` and choose "Go To Declaration" in VS Code (or whatever language server client you're using), it will take you to the type declaration for this method. Alternatively, if you hover over `add_from_file`, the client will display its type signature.

If you are using only the CLI version of pyright, you can insert the code `reveal_type(builder.add_from_file)` to get the same information that would appear in the editor when hovering.

**Comment by T0maas** [comment]
Module `gi` is from [PyGObject](https://pygobject.readthedocs.io/en/latest/) . I use pyright in coc-pyright neovim. I installed [pygobject-stubs](https://github.com/pygobject/pygobject-stubs), but problem still persists. I have same problem with pylance in vscode. 
Function `add_from_file` is defined as: 
```python
 def add_from_file(*args, **kwargs): 
```

**Comment by erictraut** [comment]
This appears to be a bug in the pyobject-stubs. The method `Builder.add_from_file` is declared as an instance method, so it should have a `self` parameter.  The same problem exists for other methods in this same class. If I add a `self` parameter, the problem goes away. I recommend filing a bug or a PR in the pygobject-stubs project.

I'll explore whether there's a straightforward way to make pyright more resilient to this sort of bug. It's the first time I've seen this problem in a type stub, so I don't think it's a common problem.

**Comment by erictraut** [comment]
I've updated pyright to handle this case better, but the problem should also be fixed in the pygobject-stubs.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.268, which I just published. It will also be included in a future version of pylance.
